oc2/ocinternal.c

```diff
@@ -385,15 +385,13 @@ static OCerror
 ocextractddsinmemory(OCstate* state, OCtree* tree, OCflags flags)
 {
     OCerror stat = OC_NOERR;
-    size_t ddslen, bod, bodfound;
+    size_t ddslen, bod;
     /* Read until we find the separator (or EOF)*/
-    bodfound = ocfindbod(state->packet,&bod,&ddslen);
-    if(!bodfound) {/* No BOD; pretend */
-	bod = tree->data.bod;
-	ddslen = tree->data.datasize;
+    int bodfound = ocfindbod(state->packet,&bod,&ddslen);
```

Comment:
@DennisHeimbigner does inverting this logic work for you? I'm less familiar immediately with what is going on here.